### PR TITLE
Checkbox - Add new focus styles

### PR DESCRIPTION
**Pull request by hepiyellow**
- Make checkbox focusable
- Add focus state
- Add focus styles

### CODE REVIEW

File: src/Checkbox/Checkbox.driver.js [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
 import labelDriverFactory from '../Label/Label.driver';
 import {testkitFactoryCreator} from '../test-common';
 
 const labelTestkitFactory = testkitFactoryCreator(labelDriverFactory);
 
-const checkboxDriverFactory = ({element, wrapper, component}) => {
+const checkboxDriverFactory = ({element, wrapper, component, eventTrigger}) => {
 
-  const checkbox = $(element).find('input')[0];
+  const input = $(element).find('input')[0];
```
Comment (liorzisman) [inline src/Checkbox/Checkbox.driver.js]: the factory provides you with the `element`, probably better to use `querySelector`on it to get the sub elements.

File: src/Checkbox/Checkbox.e2e.js [hunk 2]
```diff
@@ -6,18 +6,45 @@ describe('Checkbox', () => {
...
   });
 
-  eyes.it('should toggle state when clicked', () => {
-    waitForVisibilityOf(checkboxDriver.element(), 'Cannot find Checkbox')
-      .then(() => {
-        autoExampleDriver.setProps({checked: true});
-        expect(checkboxDriver.isChecked()).toBe(true);
+  // This test doesn't need eyes, it is only to verify that
+  // checkboxDriver.isChecked() works.
+  it('should set unchecked when clicked', async () => {
```
Comment (liorzisman) [inline src/Checkbox/Checkbox.e2e.js]: `isChecked` basically invokes protractor's `isSelected`. assuming we trust protractor, there is no need for this test in my opinion but not a must

File: src/Checkbox/Checkbox.e2e.js [hunk 3]
```diff
@@ -6,18 +6,45 @@ describe('Checkbox', () => {
...
+  });
 
-        autoExampleDriver.setProps({checked: false});
-        expect(checkboxDriver.isChecked()).toBe(false);
-      });
+  eyes.it('should show focused styles', async () => {
+    await clickTab();
   });
+
+  describe('has error', () => {
```
Comment (liorzisman) [inline src/Checkbox/Checkbox.e2e.js]: i would assert the error/focus in a way @shlomitc

File: src/Checkbox/Checkbox.js [hunk 4]
```diff
@@ -38,6 +44,14 @@ class Checkbox extends WixComponent {
 
   _id = `${Checkbox.displayName}-${uniqueId()}`;
 
+  handleInputBlur = () => {
+    this.state.isFocused && this.setState({isFocused: false});
```
Comment (liorzisman) [inline src/Checkbox/Checkbox.js]: probably not saving much with the `this.state.isFocused` before the `setState`

File: src/Checkbox/Checkbox.e2e.js (hunk 2 above)
Comment (hepiyellow) [reply src/Checkbox/Checkbox.e2e.js]: It is not testing that `isSelect` is working. It is testing that `isChecked` is implemented correctly, and not for example, returns always `true`. I'll fix the comment.

**Comment by hepiyellow** [comment]
Failed tests are e2e snapshots, which can be approved.

File: src/Checkbox/Checkbox.js [hunk 5]
```diff
@@ -77,8 +93,13 @@ class Checkbox extends WixComponent {
           />
 
         <Label for={id} appearance={disabled ? 'T1.4' : 'T1.1'} dataHook="checkbox-label">
-          <div className={classNames(styles.checkbox, styles[size])}>
-            <div className={styles.inner}>
+          <div
+            className={classNames(styles.checkbox, styles[size])}
+            tabIndex={disabled ? -1 : 0}
```
Comment (alisey) [inline src/Checkbox/Checkbox.js]: Should be null instead of -1

File: src/Checkbox/Checkbox.driver.js (hunk 1 above)
Comment (argshook) [reply src/Checkbox/Checkbox.driver.js]: 3.0.0 release includes removed jquery from drivers, so it would be better not to add $ here

**Pull request merged**